pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -350,8 +351,8 @@ func (c *Cluster) makeDeployment(osdProps osdProperties, osd OSDInfo, provisionC
 			InitContainers:     initContainers,
 			Containers: []v1.Container{
 				{
-					Command:         command,
-					Args:            args,
+					Command:         []string{"/bin/sh"},
```

Comment:
Let's not change this on the osd daemon container. We don't want to risk the long-running osd command. For example, the `ceph-osd` command really should be PID 1 or else the pod shutdown will be affected. When you change the command to `/bin/sh`, it changes the whole PID 1 behavior.

Reply:
thanks for pointing this out. But then how do you suggest to go ahead because if we want to go with ceph-osd being PID 0 , then one of the things which we need to do is to run ceph-osd as a PID 0 which will essentially nullify our attempt at running sleep infinity.

Reply:
The goal of this PR is to pause before starting `ceph-osd`, right? The right place to make this change will be in an init container. The init containers must complete before the `ceph-osd` container is started. Then if the pause container never completes, the `ceph-osd` container will never start.

Reply:
sure @travisn . 
As you pointed out the goal of this PR is to pause before starting ceph-osd. Only thing is that one might not get the exact same ENV with the init container as with the ceph-osd , so i am honestly not really sure if that is something we should thrive for or no. I think you would be able to answer that question in the best way.

Reply:
@mridulv We actually already have an init container with a script. This is the init container that activates the OSD config. At the end of the script you could add your pause based on the env var. See the [init container script](https://github.com/rook/rook/blob/f6b7aac3b62fe625fa92a590a6f6d1eb907c7fbc/pkg/operator/ceph/cluster/osd/spec.go#L49-L92).

Reply:
I have made the changes, let me know if this works.

Reply:
LGTM, this turned out nice and simple. Were you able to confirm that it pauses when the var is changed in a test cluster?